dataset/database.py

```diff
@@ -81,39 +81,38 @@ def _run_on_connect(dbapi_con, con_record):
         self._tables = {}
 
     @property
-    def executable(self):
+    def conn(self):
         """Connection against which statements will be executed."""
-        with self.lock:
-            tid = threading.get_ident()
-            if tid not in self.connections:
-                self.connections[tid] = self.engine.connect()
-            return self.connections[tid]
+        try:
+            return self.local.conn
+        except AttributeError:
+            self.local.conn = self._engine.connect()
+            self.local.conn.begin()
```

Comment:
it seems calling `conn.begin()` here means that there is only ever one transaction opened, even in presence of multiple `table.insert()` calls. 

IMHO it would be better to have an explicit semantic where `db.conn` just returns the connection and transaction context is handled by the respective operations (insert, etc.). There should be a distinction between the autocommit case (user did not request a transaction) and the explicit commit case  (user started the transaction).